### DISCUSSION TITLE
Explain Image functionality

### DISCUSSION THREAD

**Discussion by leventmolla** [MacPaw/OpenAI#147] [Ideas]
OpenAI has a new model gp4-1106-vision-preview which can explain a collection of images. I think it could be done with the general chat completions endpoint, but the documentation is not very clear about the message structure. There should be a text prompt initially describing the task and follow-up messages that contain the images. I tried this and passed base64-encoded images, got errors (for some reason the number of tokens requested is a very large number and the query fails). I then tried to pass the URLs of the files for the images, which failed as well. So I am at a loss about how to use this functionality.

**Comment by kalafus** [comment]
please see [issue report 174](https://github.com/MacPaw/OpenAI/issues/174) for an example. support was just added, and is being fixed by this [pull](https://github.com/MacPaw/OpenAI/pull/175)

**Comment by nezhyborets** [comment]
There seems to have been a progress in this since the issues was created, specifically https://github.com/MacPaw/OpenAI/pull/175
Will close until someone has more relevant issues
